atlasdb-commons/src/main/java/com/palantir/common/concurrent/CoalescingSupplier.java

```diff
@@ -40,16 +45,30 @@ public CoalescingSupplier(Supplier<T> delegate) {
 
     @Override
     public T get() {
+        try {
+            return getAsync().get();
```

Comment:
a consequence of this is that if the get() method is used, we will still always execute on a thread pool.

Reply:
There's a refactor we could do where the old method still remains, but behaves differently. I don't know it's important, though.